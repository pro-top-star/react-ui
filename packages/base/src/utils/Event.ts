import { ReactNode } from 'react';
import { HTMLEvent } from '../interfaces/HTMLEvent';

export interface Parameters {
  [key: string]: number | string | object | boolean | Parameters;
}

export class Event {
  readonly source: ReactNode;
  readonly originalEvent: HTMLEvent;
  readonly parameters: any;
  readonly htmlSource: EventTarget;

  /**
   * Event Constructor
   * @param source {ReactNode} The React Element instance
   * @param originalEvent {HTMLEvent} original HTML Event that triggered this event
   * @param parameters {Parameters} event payload
   */
  constructor(source: ReactNode, originalEvent: HTMLEvent, parameters: Parameters = {}) {
    this.source = source;
    this.originalEvent = originalEvent;
    this.parameters = parameters;
    this.htmlSource = originalEvent.target;
  }

  /**
   * Static initializer for an Event
   * @param source {ReactNode} The React Element instance
   * @param originalEvent {HTMLEvent} original HTML Event that triggered this event
   * @param parameters {Parameters} event payload
   */
  public static of(source: ReactNode, originalEvent: HTMLEvent, parameters: Parameters = {}) {
    return new Event(source, originalEvent, parameters);
  }

  public getSource(): ReactNode {
    return this.source;
  }

  public getOriginalEvent(): HTMLEvent {
    return this.originalEvent;
  }

  public getParameters(): Parameters {
    return this.parameters;
  }

  public getParameter(parameter: string): any {
    return this.parameters[parameter];
  }

  public getHtmlSourceElement(): EventTarget {
    return this.htmlSource;
  }

  public cancel(): void {
    this.originalEvent.stopPropagation();
  }

  public preventDefault(): void {
    this.originalEvent.preventDefault();
  }
}
