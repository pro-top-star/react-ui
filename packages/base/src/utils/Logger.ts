/* eslint-disable no-console,no-unused-expressions */

const pad0 = (i: any, w: number) => `000${String(i)}`.slice(-w);

export enum LOG_LEVEL {
  FATAL = 0,
  ERROR = 1,
  WARNING = 2,
  INFO = 3,
  DEBUG = 4,
  TRACE = 5
}

class DeviceLogger {
  public defaultComponent: string;
  public sWindowName = window.top === window ? '' : `[${window.location.pathname.split('/').slice(-1)[0]}] `;

  constructor(defaultComponent = 'DEVICE') {
    this.log(LOG_LEVEL.INFO, 'Device API logging initialized');
    this.defaultComponent = defaultComponent;
    return this;
  }

  public for(defaultComponent: string) {
    return new DeviceLogger(defaultComponent);
  }

  public log(iLevel: LOG_LEVEL, sMessage: string, sComponent?: string) {
    sComponent = sComponent || this.defaultComponent || '';
    const oNow = new Date();
    const oLogEntry = {
      time: `${pad0(oNow.getHours(), 2)}:${pad0(oNow.getMinutes(), 2)}:${pad0(oNow.getSeconds(), 2)}`,
      date: `${pad0(oNow.getFullYear(), 4)}-${pad0(oNow.getMonth() + 1, 2)}-${pad0(oNow.getDate(), 2)}`,
      timestamp: oNow.getTime(),
      level: iLevel,
      message: sMessage || '',
      component: sComponent || ''
    };
    if (window.console) {
      const logText = `${oLogEntry.date} ${oLogEntry.time} ${this.sWindowName}${oLogEntry.message} - ${oLogEntry.component}`;
      switch (iLevel) {
        case LOG_LEVEL.FATAL:
        case LOG_LEVEL.ERROR:
          console.error(logText);
          break;
        case LOG_LEVEL.WARNING:
          console.warn(logText);
          break;
        case LOG_LEVEL.INFO:
          console.info ? console.info(logText) : console.log(logText);
          break; // info not available in iOS simulator
        case LOG_LEVEL.DEBUG:
          console.debug ? console.debug(logText) : console.log(logText);
          break; // debug not available in IE, fallback to log
        case LOG_LEVEL.TRACE:
          console.trace ? console.trace(logText) : console.log(logText);
          break; // trace not available in IE, fallback to log
      }
    }
    return oLogEntry;
  }

  //  Public API
  public fatal(msg: string, component?: string) {
    return this.log(LOG_LEVEL.FATAL, msg, component);
  }

  public error(msg: string, component?: string) {
    return this.log(LOG_LEVEL.ERROR, msg, component);
  }

  public warning(msg: string, component?: string) {
    return this.log(LOG_LEVEL.WARNING, msg, component);
  }

  public info(msg: string, component?: string) {
    return this.log(LOG_LEVEL.INFO, msg, component);
  }

  public debug(msg: string, component?: string) {
    return this.log(LOG_LEVEL.DEBUG, msg, component);
  }

  public trace(msg: string, component?: string) {
    return this.log(LOG_LEVEL.TRACE, msg, component);
  }
}

export const Logger = new DeviceLogger();
