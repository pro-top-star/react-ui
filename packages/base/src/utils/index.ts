import { UIEvent } from 'react';

export const deprecationNotice = (component: string, message: string) => {
  if (process.env.NODE_ENV === 'development') {
    const value = `*** ui5-webcomponents-react Deprecation Notice - ${component} ***\n`;
    // eslint-disable-next-line no-console
    console.warn(`${value}${message}`);
  }
};

export const getScrollBarWidth = () => {
  const inner = document.createElement('p');
  inner.style.width = '100%';
  inner.style.height = '200px';

  const outer = document.createElement('div');
  outer.style.position = 'absolute';
  outer.style.top = '0px';
  outer.style.left = '0px';
  outer.style.visibility = 'hidden';
  outer.style.width = '200px';
  outer.style.height = '150px';
  outer.style.overflow = 'hidden';
  outer.appendChild(inner);

  document.body.appendChild(outer);
  const w1 = inner.offsetWidth;
  outer.style.overflow = 'scroll';
  let w2 = inner.offsetWidth;

  if (w1 === w2) {
    w2 = outer.clientWidth;
  }

  document.body.removeChild(outer);
  return w1 - w2;
};

export const enrichEventWithDetails = <T = {}>(event: UIEvent, payload: T = {} as any) => {
  if (event.hasOwnProperty('persist')) {
    // if there is a persist method, it's an SyntheticEvent so we need to persist it
    event.persist();
  }

  const shouldCreateNewDetails =
    event.detail === null || event.detail === undefined || typeof event.detail !== 'object';
  Object.defineProperty(event, 'detail', {
    value: shouldCreateNewDetails ? {} : event.detail,
    writable: true,
    configurable: true
  });
  Object.assign(event.detail, payload);
  return (event as unknown) as CustomEvent<T>;
};
