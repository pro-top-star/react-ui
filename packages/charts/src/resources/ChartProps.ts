export const labels = ['Label 1', 'Label 2', 'Label 3'];
export const singleDataset = [
  {
    label: 'Dataset 1',
    data: [1, 2, 3]
  }
];

export const datasets = [
  singleDataset[0],
  {
    label: 'Dataset 2',
    data: [4, 5, 6]
  }
];
