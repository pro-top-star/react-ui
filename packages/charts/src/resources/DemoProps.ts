export const complexDataSet = [
  {
    name: 'January',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'February',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'March',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'April',
    users: 280,
    sessions: 80,
    volume: 604
  },
  {
    name: 'May',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'June',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'July',
    users: 20,
    sessions: 470,
    volume: 450
  },
  {
    name: 'August',
    users: 220,
    sessions: 180,
    volume: 104
  },
  {
    name: 'September',
    users: 200,
    sessions: 360,
    volume: 879
  },
  {
    name: 'October',
    users: 250,
    sessions: 500,
    volume: 200
  },
  {
    name: 'November',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'December',
    users: 280,
    sessions: 80,
    volume: 604
  }
];

export const simpleDataSet = [
  {
    name: 'January / Month',
    users: 100
  },
  {
    name: 'February / Month',
    users: 230
  },
  {
    name: 'March / Month',
    users: 240
  },
  {
    name: 'April / Month',
    users: 280
  },
  {
    name: 'May / Month',
    users: 100
  },
  {
    name: 'June / Month',
    users: 230
  },
  {
    name: 'July / Month',
    users: 20
  },
  {
    name: 'August / Month',
    users: 220
  },
  {
    name: 'September / Month',
    users: 200
  },
  {
    name: 'October / Month',
    users: 250
  },
  {
    name: 'November / Month',
    users: 240
  },
  {
    name: 'December / Month',
    users: 280
  }
];

export const secondaryDimensionDataSet = [
  {
    name: 'January / Month',
    users: 100,
    dimension: 'C4S Components'
  },
  {
    name: 'February / Month',
    users: 230,
    dimension: 'C4S Components'
  },
  {
    name: 'March / Month',
    users: 240,
    dimension: 'C4S Components'
  },
  {
    name: 'April / Month',
    users: 280,
    dimension: 'C4S Components'
  },
  {
    name: 'May / Month',
    users: 100,
    dimension: 'C4S Components'
  },
  {
    name: 'June / Month',
    users: 230,
    dimension: 'C4S Components'
  },
  {
    name: 'July / Month',
    users: 20,
    dimension: 'Harmnony Insight'
  },
  {
    name: 'August / Month',
    users: 220,
    dimension: 'Harmnony Insight'
  },
  {
    name: 'September / Month',
    users: 200,
    dimension: 'Harmnony Insight'
  },
  {
    name: 'October / Month',
    users: 250,
    dimension: 'Intelligent Workplace'
  },
  {
    name: 'November / Month',
    users: 240,
    dimension: 'Intelligent Workplace'
  },
  {
    name: 'December / Month',
    users: 280,
    dimension: 'Pipeline Health Manager'
  }
];

export const secondaryDimensionComposedDataSet = [
  {
    name: 'January',
    users: 100,
    sessions: 300,
    volume: 756,
    dimension: 'Harmnony Insight'
  },
  {
    name: 'February',
    users: 230,
    sessions: 330,
    volume: 880,
    dimension: 'Harmnony Insight'
  },
  {
    name: 'March',
    users: 240,
    sessions: 404,
    volume: 700,
    dimension: 'C4S Components'
  },
  {
    name: 'April',
    users: 280,
    sessions: 80,
    volume: 604,
    dimension: 'C4S Components'
  },
  {
    name: 'May',
    users: 100,
    sessions: 300,
    volume: 756,
    dimension: 'C4S Components'
  },
  {
    name: 'June',
    users: 230,
    sessions: 330,
    volume: 880,
    dimension: 'Intelligent Workplace'
  },
  {
    name: 'July',
    users: 20,
    sessions: 470,
    volume: 450,
    dimension: 'Intelligent Workplace'
  },
  {
    name: 'August',
    users: 220,
    sessions: 180,
    volume: 104,
    dimension: 'Intelligent Workplace'
  },
  {
    name: 'September',
    users: 200,
    sessions: 360,
    volume: 879,
    dimension: 'Intelligent Workplace'
  },
  {
    name: 'October',
    users: 250,
    sessions: 500,
    volume: 200,
    dimension: 'Pipeline Health Manager'
  },
  {
    name: 'November',
    users: 240,
    sessions: 404,
    volume: 700,
    dimension: 'Pipeline Health Manager'
  },
  {
    name: 'December',
    users: 280,
    sessions: 80,
    volume: 604,
    dimension: 'Pipeline Health Manager'
  }
];

export const bigDataSet = [
  {
    name: 'January',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'February',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'March',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'April',
    users: 280,
    sessions: 80,
    volume: 604
  },
  {
    name: 'May',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'June',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'July',
    users: 20,
    sessions: 470,
    volume: 450
  },
  {
    name: 'August',
    users: 220,
    sessions: 180,
    volume: 104
  },
  {
    name: 'September',
    users: 200,
    sessions: 360,
    volume: 879
  },
  {
    name: 'October',
    users: 250,
    sessions: 500,
    volume: 200
  },
  {
    name: 'November',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'December',
    users: 280,
    sessions: 80,
    volume: 604
  },
  {
    name: 'January',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'February',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'March',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'April',
    users: 280,
    sessions: 80,
    volume: 604
  },
  {
    name: 'May',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'June',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'July',
    users: 20,
    sessions: 470,
    volume: 450
  },
  {
    name: 'August',
    users: 220,
    sessions: 180,
    volume: 104
  },
  {
    name: 'September',
    users: 200,
    sessions: 360,
    volume: 879
  },
  {
    name: 'October',
    users: 250,
    sessions: 500,
    volume: 200
  },
  {
    name: 'November',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'December',
    users: 280,
    sessions: 80,
    volume: 604
  },
  {
    name: 'January',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'February',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'March',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'April',
    users: 280,
    sessions: 80,
    volume: 604
  },
  {
    name: 'May',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'June',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'July',
    users: 20,
    sessions: 470,
    volume: 450
  },
  {
    name: 'August',
    users: 220,
    sessions: 180,
    volume: 104
  },
  {
    name: 'September',
    users: 200,
    sessions: 360,
    volume: 879
  },
  {
    name: 'October',
    users: 250,
    sessions: 500,
    volume: 200
  },
  {
    name: 'November',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'December',
    users: 280,
    sessions: 80,
    volume: 604
  },
  {
    name: 'January',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'February',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'March',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'April',
    users: 280,
    sessions: 80,
    volume: 604
  },
  {
    name: 'May',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'June',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'July',
    users: 20,
    sessions: 470,
    volume: 450
  },
  {
    name: 'August',
    users: 220,
    sessions: 180,
    volume: 104
  },
  {
    name: 'September',
    users: 200,
    sessions: 360,
    volume: 879
  },
  {
    name: 'October',
    users: 250,
    sessions: 500,
    volume: 200
  },
  {
    name: 'November',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'December',
    users: 280,
    sessions: 80,
    volume: 604
  },
  {
    name: 'January',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'February',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'March',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'April',
    users: 280,
    sessions: 80,
    volume: 604
  },
  {
    name: 'May',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'June',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'July',
    users: 20,
    sessions: 470,
    volume: 450
  },
  {
    name: 'August',
    users: 220,
    sessions: 180,
    volume: 104
  },
  {
    name: 'September',
    users: 200,
    sessions: 360,
    volume: 879
  },
  {
    name: 'October',
    users: 250,
    sessions: 500,
    volume: 200
  },
  {
    name: 'November',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'December',
    users: 280,
    sessions: 80,
    volume: 604
  },
  {
    name: 'January',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'February',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'March',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'April',
    users: 280,
    sessions: 80,
    volume: 604
  },
  {
    name: 'May',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'June',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'July',
    users: 20,
    sessions: 470,
    volume: 450
  },
  {
    name: 'August',
    users: 220,
    sessions: 180,
    volume: 104
  },
  {
    name: 'September',
    users: 200,
    sessions: 360,
    volume: 879
  },
  {
    name: 'October',
    users: 250,
    sessions: 500,
    volume: 200
  },
  {
    name: 'November',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'December',
    users: 280,
    sessions: 80,
    volume: 604
  },
  {
    name: 'January',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'February',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'March',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'April',
    users: 280,
    sessions: 80,
    volume: 604
  },
  {
    name: 'May',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'June',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'July',
    users: 20,
    sessions: 470,
    volume: 450
  },
  {
    name: 'August',
    users: 220,
    sessions: 180,
    volume: 104
  },
  {
    name: 'September',
    users: 200,
    sessions: 360,
    volume: 879
  },
  {
    name: 'October',
    users: 250,
    sessions: 500,
    volume: 200
  },
  {
    name: 'November',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'December',
    users: 280,
    sessions: 80,
    volume: 604
  },
  {
    name: 'January',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'February',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'March',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'April',
    users: 280,
    sessions: 80,
    volume: 604
  },
  {
    name: 'May',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'June',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'July',
    users: 20,
    sessions: 470,
    volume: 450
  },
  {
    name: 'August',
    users: 220,
    sessions: 180,
    volume: 104
  },
  {
    name: 'September',
    users: 200,
    sessions: 360,
    volume: 879
  },
  {
    name: 'October',
    users: 250,
    sessions: 500,
    volume: 200
  },
  {
    name: 'November',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'December',
    users: 280,
    sessions: 80,
    volume: 604
  }
];
