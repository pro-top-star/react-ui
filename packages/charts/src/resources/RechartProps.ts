export const label = 'name';

export const dataset = [
  {
    name: 'January',
    users: 100,
    sessions: 300,
    volume: 756
  },
  {
    name: 'February',
    users: 230,
    sessions: 330,
    volume: 880
  },
  {
    name: 'March',
    users: 240,
    sessions: 404,
    volume: 700
  },
  {
    name: 'April',
    users: 280,
    sessions: 80,
    volume: 604
  }
];
