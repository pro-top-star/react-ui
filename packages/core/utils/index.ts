export * from './Device';
export * from './isNumeric';
