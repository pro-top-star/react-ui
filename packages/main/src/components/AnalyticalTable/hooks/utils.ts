export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
