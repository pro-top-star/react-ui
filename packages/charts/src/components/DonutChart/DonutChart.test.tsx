import { mountThemedComponent, renderThemedComponent } from '@shared/tests/utils';
import * as React from 'react';
import { labels, singleDataset } from '../../test/resources/ChartProps';
import { DonutChart } from './index';

describe('DonutChart', () => {
  test('Renders with data', () => {
    renderThemedComponent(<DonutChart labels={labels} datasets={singleDataset} />);
  });

  test('loading placeholder', () => {
    const wrapper = mountThemedComponent(<DonutChart labels={labels} datasets={[]} loading />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
