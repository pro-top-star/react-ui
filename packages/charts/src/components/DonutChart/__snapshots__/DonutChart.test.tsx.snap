// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonutChart loading placeholder 1`] = `
<div
  class="ChartContainer--chart- ChartContainer--chart-d0-"
  style="position: relative; padding-top: 6px; width: 300px; height: 300px;"
>
  <svg
    aria-label="Loading interface..."
    preserveAspectRatio="none"
    role="img"
    viewBox="0 0 150 150"
  >
    <title>
      Loading interface...
    </title>
    <rect
      clip-path="CLIP-PATH-URL"
      height="150"
      style="fill: url(#STYLE-URL);"
      width="150"
      x="0"
      y="0"
    />
    <defs>
      <clipPath
        id="CLIP-PATH-URL"
      >
        <circle
          cx="85"
          cy="75"
          r="60"
        />
      </clipPath>
      <linearGradient
        id="STYLE-URL"
      >
        <stop
          offset="0%"
          stop-color="#6a6d70"
          stop-opacity="0.2"
        >
          <animate
            attributeName="offset"
            dur="2s"
            keyTimes="0; 0.25; 1"
            repeatCount="indefinite"
            values="-2; -2; 1"
          />
        </stop>
        <stop
          offset="50%"
          stop-color="#d9d9d9"
          stop-opacity="1"
        >
          <animate
            attributeName="offset"
            dur="2s"
            keyTimes="0; 0.25; 1"
            repeatCount="indefinite"
            values="-1; -1; 2"
          />
        </stop>
        <stop
          offset="100%"
          stop-color="#6a6d70"
          stop-opacity="0.2"
        >
          <animate
            attributeName="offset"
            dur="2s"
            keyTimes="0; 0.25; 1"
            repeatCount="indefinite"
            values="0; 0; 3"
          />
        </stop>
      </linearGradient>
    </defs>
  </svg>
  <div
    class="legend"
  />
</div>
`;
